jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/ReopenCaseCommand.java

```diff
@@ -79,20 +79,22 @@ public Void execute(Context context) {
         
         caseEventSupport.fireBeforeCaseReopened(caseId, caseFile, deploymentId, caseDefinitionId, data);
         
-        logger.debug("Updating case file in working memory");
+        logger.debug("Removing case file from working memory to allow refiring of rules");
         FactHandle factHandle = ksession.getFactHandle(caseFile);
+        ksession.delete(factHandle);
         ((CaseFileInstanceImpl)caseFile).setCaseReopenDate(new Date());
         if (data != null && !data.isEmpty()) {
             caseFile.addAll(data);
         }
-        ksession.update(factHandle, caseFile);
+        
         
         logger.debug("Starting process instance for case {} and case definition {}", caseId, caseDefinitionId);
         CorrelationKey correlationKey = correlationKeyFactory.newCorrelationKey(caseId);
         Map<String, Object> params = new HashMap<>();
         // set case id to allow it to use CaseContext when creating runtime engine
         params.put(EnvironmentName.CASE_ID, caseId);
         long processInstanceId = processService.startProcess(deploymentId, caseDefinitionId, correlationKey, params);
```

Comment:
Hi @mswiderski 
There are 3 tests failing in _CaseServiceImplTest_ class:

_testStartThenReopenEmptyCase
testStartThenReopenEmptyCaseUpdateData
testStartThenReopenEmptyCaseUpdateDescription_

_expected:<[my first case]> but was:<[#{name}]>_

It seems like when starting the process, {name} variable has no value

Reply:
working on a fix, thanks for checking on it

Reply:
pushed a fix so let's see how it goes in Jenkins now...

Reply:
I've tested that approach, but some assertions are failing, because expecting 3 variables instead of 4 (because this time -inserting it like that- doesn't have caseId_ prefix):
variableId=caseFile_name  vs. variableId=name

Reply:
And if add them with the prefix
for (Map.Entry<String, Object> entry : caseData.entrySet()) {
            params.put(VariableScope.CASE_FILE_PREFIX + entry.getKey(), entry.getValue());
        }
Then, getVariable at VariableScopeInstance class, don't transform it as
caseFiles.size() == 0

Reply:
you're right, I refactored this to suit both needs, have a look again

Reply:
thanks Maciej! It seems now it's working (though let Jenkins to finish)